appium/webdriver/webdriver.py

```diff
@@ -142,6 +148,24 @@ def __init__(self, command_executor='http://127.0.0.1:4444/wd/hub',
         By.IMAGE = MobileBy.IMAGE
         By.CUSTOM = MobileBy.CUSTOM
 
+    def _update_command_executor(self, keep_alive):
```

Comment:
should probably check that these values actually exist. just because someone has set the `direct_connection` param doesn't mean the server supports it, right?

up to you whether you think we should throw an error if one of these values is not present, or just silently ignore the request to direct connect.

Reply:
nice catch, I forgot it...

I'll show warning or error message rather than stopping test script since whether a server cannot handle the following requests or not depend on them.